lib/libc/string/wcsncmp.c

```diff
@@ -38,22 +38,21 @@ __RCSID("$NetBSD: wcsncmp.c,v 1.3 2001/01/05 12:13:13 itojun Exp $");
 #endif
 __FBSDID("$FreeBSD$");
 
+#include <stdint.h>
 #include <wchar.h>
 
 int
 wcsncmp(const wchar_t *s1, const wchar_t *s2, size_t n)
 {
 
-	if (n == 0)
-		return (0);
-	do {
-		if (*s1 != *s2++) {
+	for (; n; --n) {
+		if (*s1 != *s2) {
 			/* XXX assumes wchar_t = int */
```

Comment:
This comment can also be removed. The code assumes that uchar is unsigned and the size of an int before, but after not so much. Same question / comment comment as above.